samples/Gallery/Pages/NotificationsPage.fs

```diff
@@ -31,16 +31,47 @@ module NotificationsPage =
         | ShowManagedNotification
         | ShowCustomManagedNotification
         | ShowNativeNotification
+        | ShowAsyncCompletedNotification
+        | ShowAsyncStatusNotifications
+        | NotifyInfo of string
         | YesCommand
         | NoCommand
         | AttachedToVisualTreeChanged of VisualTreeAttachmentEventArgs
         | ControlNotificationsShow
 
-    type CmdMsg = | NoMsg
+    //TODO What is this about?
+    type CmdMsg = | NotifyAsyncCompleted | NotifyAsyncStatusUpdates
 
+    let private notifyOneAsync =
+        async {
+            do! Async.Sleep 1000
+            // TODO causes an InvalidOperationException saying "Call from invalid thread" from the Dispatcher
+            return NotifyInfo "async operation completed"
+        }
+
+    let private notifyAsyncStatusUpdates dispatch =
```

Comment:
In this case you will need to make sure the `Msg` is dispatched in the UI Thread. I think this will fix this.
```fsharp
let mutable cts: CancellationTokenSource = null

let private notifyAsyncStatusUpdates dispatch =
        if cts = null then
            cts <- new CancellationTokenSource()

            let notifications =
                async {
                    Dispatcher.UIThread.Post(fun _ -> dispatch(NotifyInfo "started"))
                    do! Async.Sleep 1000
                    Dispatcher.UIThread.Post(fun _ -> dispatch(NotifyInfo "5"))
                    do! Async.Sleep 1000
                    Dispatcher.UIThread.Post(fun _ -> dispatch(NotifyInfo "4"))
                    do! Async.Sleep 1000
                    Dispatcher.UIThread.Post(fun _ -> dispatch(NotifyInfo "3"))
                    do! Async.Sleep 1000
                    Dispatcher.UIThread.Post(fun _ -> dispatch(NotifyInfo "2"))
                    do! Async.Sleep 1000
                    Dispatcher.UIThread.Post(fun _ -> dispatch(NotifyInfo "1"))
                    do! Async.Sleep 1000
                    Dispatcher.UIThread.Post(fun _ -> dispatch(NotifyInfo "Completed"))
                }

            Async.Start(notifications, cts.Token)
        else
            cts.Cancel()
            cts.Dispose()
            cts <- null
```

Reply:
Thanks! That did the trick for when I have access to a `dispatch` function by using `Cmd.ofEffect`, but not for my other example. See follow-up task.